api/v1beta1/openstacknetconfig_webhook.go

```diff
@@ -247,14 +275,79 @@ func (r *OpenStackNetConfig) validateStaticMacReservations(old runtime.Object) e
 				if oldInstance, ok = old.(*OpenStackNetConfig); !ok {
 					return fmt.Errorf("runtime object is not an OpenStackNetConfig")
 				}
-				if currentMAC, ok := oldInstance.Spec.OVNBridgeMacMappings.StaticReservations[node].Reservations[physnet]; ok && currentMAC != mac {
+				if currentMAC, ok := oldInstance.Spec.Reservations[node].MACReservations[physnet]; ok && currentMAC != mac {
 					return fmt.Errorf("MAC address %s of node %s must not change - new MAC address %s", currentMAC, node, mac)
 				}
 			}
 		}
 
 		// if all tests pass add to reservations
-		reservations[node] = res
+		reservations[node] = OpenStackMACNodeReservation{
+			Reservations: res.MACReservations,
+		}
 	}
+
+	return nil
+}
+
+// validateStaticIPReservations - validate static IP address reservations
+func (r *OpenStackNetConfig) validateStaticIPReservations(old runtime.Object) error {
+	// fill an empty reservations map to check for uniq IP reservations
+	reservations := map[string]IPReservation{}
+
+	for node, res := range r.Spec.Reservations {
+		for netName, resIP := range res.IPReservations {
+			//
+			// check if the IP address has a valid format
+			//
+			ip := net.ParseIP(resIP)
+			if ip == nil {
+				return fmt.Errorf("IP address %s of node %s has an invalid format", resIP, node)
+			}
+
+			//
+			// check if IP matches osnet spec
+			//
+			for _, osNet := range r.Spec.Networks {
+				for _, subnet := range osNet.Subnets {
+					if subnet.Name == netName {
+						var ipnet *net.IPNet
+						if subnet.IPv4.Cidr != "" {
+							_, ipnet, _ = net.ParseCIDR(subnet.IPv4.Cidr)
+						} else {
+							_, ipnet, _ = net.ParseCIDR(subnet.IPv4.Cidr)
```

Comment:
Should this be `IPv6`?

Reply:
right, good catch, will fix it and add an ipv6 kuttl tests which should have shown this :)